arch/arm/core/aarch64/reset.S

```diff
@@ -44,18 +53,65 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__reset)
 GTEXT(__start)
 SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 
-#ifdef CONFIG_SWITCH_TO_EL1
+	/* Setup vector table */
+	adr	x19, _vector_table
+
 	switch_el x1, 3f, 2f, 1f
 3:
-	/* Disable MMU and async exceptions routing to EL1 */
-	msr	sctlr_el1, xzr
+	/* Initialize VBAR */
+	msr	vbar_el3, x19
+	isb
+
+	/* Initialize sctlr_el3 to reset value */
+	mov_imm	x1, SCTLR_EL3_RES1
+	mrs     x0, sctlr_el3
+	orr	x0, x0, x1
+	msr	sctlr_el3, x0
+	isb
+
+	/* SError, IRQ and FIQ routing enablement in EL3 */
+	mrs	x0, scr_el3
+	orr	x0, x0, #(SCR_EL3_IRQ | SCR_EL3_FIQ | SCR_EL3_EA)
+	msr	scr_el3, x0
+
+	/*
+	* Disable access traps to EL3 for CPACR, Trace, FP, ASIMD,
+	* SVE from lower EL.
+	*/
+	mov_imm	x0, CPTR_EL3_RES_VAL
+	mov_imm	x1, (CPTR_EL3_TTA | CPTR_EL3_TFP | CPTR_EL3_TCPAC)
+	bic	x0, x0, x1
+	orr	x0, x0, #(CPTR_EL3_EZ)
+	msr	cptr_el3, x0
+	isb
+
+	/* Platform specific configurations needed in EL3 */
+	bl	z_arch_el3_plat_init
+
+#ifdef CONFIG_SWITCH_TO_EL1
+	/*
+	* Zephyr entry happened in EL3. Do EL3 specific init before
+	* dropping to lower EL.
+	*/
+	/* Enable access control configuration from lower EL */
+	mrs	x0, actlr_el3
+	orr     x0, x0, #(ACTLR_EL3_L2ACTLR | ACTLR_EL3_L2ECTLR \
+			 | ACTLR_EL3_L2CTLR)
+	orr     x0, x0, #(ACTLR_EL3_CPUACTLR | ACTLR_EL3_CPUECTLR)
+	msr	actlr_el3, x0
+
+	/* Initialize sctlr_el1 to reset value */
+	mrs     x0, sctlr_el1
+	mov_imm	x1, SCTLR_EL1_RES1
+	orr	x0, x0, x1
+	msr	sctlr_el1, x0
```

Comment:
I think this can be moved under "1:". 2 reasons for that: 1) consistency: At each EL jump label(3/2/1), we initialize `sctlr_elx` and later set few bits for enabling instruction cache, alignment checks etc. 2) Assuming we directly jump to label1, we need `sctlr_el1` initialization in place.

Reply:
In standard practice ( ref: tf-a) EL3 init takes care of reset value initialization for EL1 when PE was reset to EL3. 
2) Directly jump to label1:  Implies either PE was reset to EL1 Or  another boot stage ran in EL3 before Zephyr entry.  In both the case 'RES1' setting is not necessary in EL1.

Reply:
Okay understood, EL3 is responsible for initializing `sctlr_el1` to `RES1`.